tests/unit/Types/ArrayKeyTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Reflection\Types;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @coversDefaultClass \phpDocumentor\Reflection\Types\ArrayKey
+ */
+final class ArrayKeyTest extends TestCase
+{
+    /**
+     * @covers ::__construct
+     * @covers ::__toString
+     */
+    public function testCompoundCanBeConstructedAndStringifiedCorrectly() : void
+    {
+        $this->assertSame('array-key', (string) (new ArrayKey()));
+    }
+
+    /**
+     * @uses \phpDocumentor\Reflection\Types\Compound::__construct
+     * @uses \phpDocumentor\Reflection\Types\Integer
+     * @uses \phpDocumentor\Reflection\Types\Boolean
+     *
+     * @covers ::getIterator
+     */
+    public function testCompoundCanBeIterated() : void
+    {
+        $types = [new Integer(), new Boolean()];
+
+        foreach (new Compound($types) as $index => $type) {
```

Comment:
I'm not sure if the ArrayKey should extend the AggregatedType anyway

Reply:
The docs of psalm do say that it is an `int` or a `string`, so that would reflect into an aggregated type. But I do understand your doubts. I need to think about that as well.

Reply:
fixed